### PR TITLE
Update to JupyterLab v3.1 and support RTC

### PR DESCRIPTION
**Pull request by hbcarlos**
This is a draft PR to update to JupyterLab v3.1 and add support for RTC.

Due to some attributes on the notebook's metadata (`kernelspecs` and `language_info`), we don't share the notebook's metadata by default.

### TODO:
- [x] Upgrade dependencies to 3.1.
- [x] Use metadata from shared models.
- [x] Check that the metadata from the shared model is saved to disk.
- [x] Check that the nbconvert template works.
- [x] Check that the classic notebook extension works.

### Demo:

https://user-images.githubusercontent.com/26092748/128715933-33054277-4906-4915-a3c7-f55e980b3e15.mp4

### CODE REVIEW

**Comment by hbcarlos** [comment]
We need to wait for two PRs in JupyterLab:

- [ ] [Adds shared state](https://github.com/jupyterlab/jupyterlab/pull/10795)
- [ ] [Save shared metadata](https://github.com/jupyterlab/jupyterlab/pull/10804)

Then we can remove:
https://github.com/voila-dashboards/voila-gridstack/blob/f9d99f3ef19737158a103d73aad53d889309d93c/packages/jupyterlab-gridstack/src/editor/gridstack/model.ts#L83-L84
and
https://github.com/voila-dashboards/voila-gridstack/blob/f9d99f3ef19737158a103d73aad53d889309d93c/.eslintrc.js#L38-L39

Some of the changes I made:
* Changed the default behavior of items to locked for a better user experience. The user still can move and recize them, but when moving/resizing one item, the other items don't move.
* Added a shared attribute executed, used to track if when opening Voila-Gridstack editor another client already executed the cells. Otherwise, every time a client connects, executes all cells.

**Review by jtpio — APPROVED** [review]
Thanks, just tested locally and it works great!

File: packages/jupyterlab-gridstack/src/editor/gridstack/model.ts [hunk 1]
```diff
@@ -174,14 +191,20 @@ export class GridStackModel {
    */
   set info(info: DashboardView) {
     this._info = info;
-    const data = this._context.model.metadata.get('extensions') as Record<
+    /* const data = this._context.model.metadata.get('extensions') as Record<
```
Comment (jtpio) [inline packages/jupyterlab-gridstack/src/editor/gridstack/model.ts:194]: A minor cleanup follow-up could be to remove this comment and a couple of others.

Comment (hbcarlos) [reply packages/jupyterlab-gridstack/src/editor/gridstack/model.ts:194]: Ouch! Opening an issue!

**Pull request merged**